### PR TITLE
Code quality & documentation

### LINKED ISSUE #83: Improving code quality and documentation
Complete check of all code for quality and documentation in preparation of the final submission, this includes both the Arduino and Android code.

### PR DESCRIPTION
**Pull request by ChrisAxt**
Code quality improvements as well as code documentation, most changes were merged into master through the Arduino_Unittesting due to compatibility issues.  

resolves #83 
closes #83

### CODE REVIEW

**Review by alrapal — APPROVED** [review]
Thanks for the documentation ! Makes it way easier to follow. 
As you said the PR #96 implemented a lot of your changes, and I saw that the switch was actually from this branch ! As we say in french "Rendre à César ce qui appartient à César" ! 
(Give back to Cesar what belongs to Cesar). So good work ! 😄

File: UI/app/src/main/java/com/example/pathfinder/Activities/DriverLogin.java [hunk 1]
```diff
@@ -100,15 +99,25 @@ public void onClick(View v) {
         });
     }
 
+    /**
+     * Method to check if a given name exists in the system
+     * @param username
+     * @return
```
Comment (alrapal) [inline UI/app/src/main/java/com/example/pathfinder/Activities/DriverLogin.java]: Could we add just what is return here ?

**Review by alrapal — CHANGES_REQUESTED** [review]

**Review by iachella — APPROVED** [review]
Looks good! Nice way of documenting :)

**Review by alrapal — COMMENTED** [review]
(Not sure it is necessary but what do you think about the comments below. I know we can look at the code to know it, but I think it makes it neater to put it in the return parameter

File: UI/app/src/main/java/com/example/pathfinder/Activities/DriverLogin.java [hunk 2]
```diff
@@ -100,15 +99,25 @@ public void onClick(View v) {
...
     public boolean isRegisteredUsername(String username){
         return driversProfiles.containsKey(username);
     }
-    
+
+    /**
+     * Method to check if a given username and password belong to the same account
+     * @param username
+     * @param password
+     * @return
```
Comment (alrapal) [inline UI/app/src/main/java/com/example/pathfinder/Activities/DriverLogin.java]: And here ?

File: UI/app/src/main/java/com/example/pathfinder/Activities/ThumbstickView.java [hunk 3]
```diff
@@ -107,10 +130,19 @@ public void surfaceDestroyed(@NonNull SurfaceHolder holder) {
...
+     */
     public interface ThumbstickListener {
         void onThumbstickMoved(float xPercent, float yPercent, int id);
     }
 
+    /**
+     * Method that calculates in which position the thumbstick currently is
+     * @param v
+     * @param e
+     * @return
```
Comment (alrapal) [inline UI/app/src/main/java/com/example/pathfinder/Activities/ThumbstickView.java]: here as well?

**Review by alrapal — COMMENTED** [review]
(Just pointing out where we could add explanation regarding what is returned and when)

**Comment by juliavankirk** [comment]
Looks great, thanks for working on this!

File: UI/app/src/main/java/com/example/pathfinder/Activities/DriverLogin.java (hunk 2 above)
Comment (ChrisAxt) [reply UI/app/src/main/java/com/example/pathfinder/Activities/DriverLogin.java]: Good catch, will make the changes :)

**Comment by ediz-genc** [comment]
> Code quality improvements as well as code documentation, most changes were merged into master through the Arduino_Unittesting due to compatibility issues.

**Comment by bassamEldesouki** [comment]
Seems like everything is covered now, no comment from me, Thanks for this work :)

**Comment by alrapal** [comment]
Reopen because needs to be merged and addressed

**Review by ediz-genc — APPROVED** [review]
Other than tiny changes, all documentation looks good. Good work :)

**Review by alrapal — APPROVED** [review]
Thanks for the changes ! 😄

**Pull request merged**